simulator/operators.py

```diff
@@ -56,7 +56,8 @@ def create(N, dtype=tf.complex64):
     Returns:
         Tensor([N, N], dtype): NxN creation operator
     """
-    return tf.cast(tf.linalg.adjoint(destroy(N, dtype=tf.complex64)), dtype=dtype)
+    # Preserve max precision in intermediate calculations until final cast
+    return tf.cast(tf.linalg.adjoint(destroy(N, dtype=tf.complex128)), dtype=dtype)
```

Comment:
This comment probably could be worded better, by "intermediate" it's really just when creating the position and momentum operators. This only happens once (during initialization), and then afterwards p and q are stored/used with c64.

I did this to match the QuTIP behavior exactly (otherwise there is a small difference, see `simulator/test/test_operators.py`, but it's probably not a big difference.